examples/sum_pe/isa.py

```diff
@@ -1,18 +1,26 @@
-from hwtypes import new_instruction
 from hwtypes.adt import Product, Sum, Tuple, Enum
-from functools import lru_cache
-from peak import Const
+from peak import family_closure
+from types import SimpleNamespace
 
-@lru_cache(None)
+
+class Op(Enum):
+    A = 1
+    B = 2
+
+@family_closure
 def ISA_fc(family):
     Word = family.BitVector[8]
     Bit  = family.BitVector[1]
 
-    Operand0T = Word
-    Operand1T = Sum[Word, Tuple[Word, Bit]]
-    class Inst(Product):
-        class Opcode(Enum):
-            A  = 1
-            B  = 2
-        offset = Word
-    return Inst, Operand0T, Operand1T, Word, Tuple[Word, Bit]
+    # TODO there is a bug when trying to use Product here
+    #class BitOp(Product):
+    #    op = Op
+    #    neg = Bit
+
```

Comment:
Remove commented code.  If there is a TODO document in as issue.

Reply:
Fixed